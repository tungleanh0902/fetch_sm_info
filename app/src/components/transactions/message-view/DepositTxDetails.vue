<template>
  <div class="tx__content">
    <TransactionIcon :transaction-type="type" />
    <div class="tx__content__left">
      <h3>{{ caption }}</h3>
      <span>On&nbsp;</span>
      <router-link
        :to="{
          name: 'proposal',
          params: { proposalId: String(transaction.details.proposalId) },
        }"
        >Proposal &#35;{{ transaction.details.proposalId }}</router-link
      >
    </div>
    <div class="tx__content__right">
      <p class="amount">
        {{ transaction.details.amount.amount | prettyLong }}&nbsp;
        {{ transaction.details.amount.denom }}
      </p>
    </div>
  </div>
</template>

<script>
import { prettyLong } from "scripts/num.js"
import TransactionIcon from "../TransactionIcon"

export default {
  name: `deposit-tx-details`,
  filters: {
    prettyLong,
  },
  components: {
    TransactionIcon,
  },
  props: {
    transaction: {
      type: Object,
      required: true,
    },
    validators: {
      type: Object,
      required: true,
    },
  },
  data: () => {
    return {
      type: `Deposit`,
      caption: `Deposit`,
    }
  },
}
</script>
