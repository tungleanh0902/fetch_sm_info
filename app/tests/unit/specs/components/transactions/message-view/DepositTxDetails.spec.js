import { shallowMount } from "@vue/test-utils"
import DepositTxDetails from "src/components/transactions/message-view/DepositTxDetails"

describe(`DepositTxDetails`, () => {
  let wrapper

  const tx = {
    type: "DepositTx",
    hash: "3CA728671B8078E71697B62237AD694052779F80B56880F6A6F1702F53EA3081",
    height: 308453,
    timestamp: "2020-01-10T09:02:54Z",
    memo: "",
    success: true,
    fees: [
      {
        denom: "ATOM",
        amount: "0.00045",
      },
    ],
    details: {
      proposalId: 1,
      amount: {
        denom: "ATOM",
        amount: "10",
      },
    },
  }

  it(`renders a deposit transaction message`, () => {
    wrapper = shallowMount(DepositTxDetails, {
      propsData: {
        transaction: tx,
        validators: {},
      },
      stubs: [`router-link`],
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
