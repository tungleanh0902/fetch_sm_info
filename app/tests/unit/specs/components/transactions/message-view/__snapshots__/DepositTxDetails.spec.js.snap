// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositTxDetails renders a deposit transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiontype="Deposit"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Deposit
    </h3>
     
    <span>
      On 
    </span>
     
    <router-link-stub
      to="[object Object]"
    >
      Proposal #1
    </router-link-stub>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      10 
      ATOM
    
    </p>
  </div>
</div>
`;
